periodic_tasks/pools/tasks.py

```diff
@@ -0,0 +1,156 @@
+        logger.info(
+            'Processing %s %s for vault %s. Wallet address: %s ...',
+            Web3.from_wei(base_to_swap, 'ether'),
+            NETWORK_BASE_TICKERS[NETWORK],
+            pool.vault_address,
+            pool.wallet.address,
+        )
+
+        if base_to_swap < network_config.MIN_POOL_SWAP_AMOUNT:
+            logger.info('Distributed amount too small, skipping vault %s...', pool.vault_address)
+            continue
+
+        # wrap native eth for mainnet
+        if NETWORK == MAINNET:
+            base_to_swap = await _convert_to_weth(wallet=pool.wallet, amount=base_to_swap)
+
+        swapped_amount = await CowProtocolWrapper().swap(
+            wallet=pool.wallet,
+            sell_token=pool.swap_from_token,
+            buy_token=pool.swap_to_token,
+            sell_amount=base_to_swap,
+        )
+        if swapped_amount:
+            logger.info(
+                'Swapped %s %s via CowSwap for vault %s...',
+                Web3.from_wei(swapped_amount, 'ether'),
+                pool.ticker,
+                pool.vault_address,
+            )
+
+        else:
+            logger.info('Can\'t swap tokens via CowSwap, skipping vault %s...', pool.vault_address)
+            continue
+
+        if pool.ticker == USDS_TICKER:
+            await _convert_to_susds(pool.wallet)
+            logger.info('Successfully converted usds to susds for vault %s...', pool.vault_address)
+
+        distributed_token_contract = get_erc20_contract(
+            address=pool.distributed_token,
+        )
+        distributed_token_amount = await distributed_token_contract.get_balance(pool.wallet.address)
+        logger.info(
+            'Distributing %s %s for vault %s...',
+            Web3.from_wei(distributed_token_amount, 'ether'),
+            pool.ticker,
+            pool.vault_address,
+        )
+
+        await distribute_tokens(
+            token=pool.distributed_token,
+            amount=distributed_token_amount,
+            vault_address=pool.vault_address,
+        )
+        logger.info('Vault %s was processed successfully')
+
+
+def _build_pool_settings() -> list[PoolSettings]:
+    pool_settings: list[PoolSettings] = []
+    for ticker in NETWORK_TICKERS[NETWORK]:
+        wallet = TICKER_TO_SETTINGS[ticker][0]
+        vault_address = TICKER_TO_SETTINGS[ticker][1]
+        if not wallet:
+            continue
+        if not vault_address:
+            logger.warning('Missed vault environment variable for ticker %s', ticker)
+            continue
+        vault_address = Web3.to_checksum_address(vault_address)
+        pool_settings.append(
+            PoolSettings(
+                ticker=ticker,
+                wallet=wallet,
+                vault_address=vault_address,
+            )
+        )
+
+    return pool_settings
+
+
+async def _convert_to_weth(wallet: LocalAccount, amount: Wei) -> Wei:
+    """Convert ETH to WETH"""
+    converted_amount = Wei(amount - MIN_ETH_FOR_GAS_AMOUNT)
```

Comment:
ideally we should convert everything we've received from the pool. The gas amount should be funded from a separate acc.